applications/FluidDynamicsApplication/python_scripts/cfl_output_process.py

```diff
@@ -0,0 +1,180 @@
+# Importing the Kratos Library
+import KratosMultiphysics
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+
+# other imports
+from KratosMultiphysics.time_based_ascii_file_writer_utility import TimeBasedAsciiFileWriterUtility
+
+from statistics import mean, stdev
+
+
+def Factory(settings, model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception(
+            "expected input shall be a Parameters object, encapsulating a json string")
+
+    return CFLOutputProcess(model, settings["Parameters"])
+
+
+class CFLOutputProcess(KratosMultiphysics.Process):
+    """
+    Auxiliary base class to output total flow forces
+    over obstacles in fluid dynamics problems.
+    A derived class needs to be implemented to be able to use
+    this functionality, as calling the base class alone is not enough.
+    """
+
+    def __init__(self, model, params):
+        """
+        Auxiliary class to output total flow forces over obstacles
+        in fluid dynamics problems for a body fitted model part.
+        """
+        KratosMultiphysics.Process.__init__(self)
+
+        default_settings = KratosMultiphysics.Parameters("""
+            {
+                "model_part_name"      : "",
+                "interval"             : [0.0, 1e30],
+                "cfl_threshold"        : 5.0,
+                "print_to_screen"      : false,
+                "print_format"         : ".8f",
+                "write_output_file"    : true,
+                "output_step"          : 8,
+                "output_file_settings": {}
+            }
+            """)
+
+        self.params = params
+        # Detect "End" as a tag and replace it by a large number
+        if(self.params.Has("interval")):
+            if(self.params["interval"][1].IsString()):
+                if(self.params["interval"][1].GetString() == "End"):
+                    self.params["interval"][1].SetDouble(1e30)
+                else:
+                    raise Exception("The second value of interval can be \"End\" or a number, interval currently:" +
+                                    self.params["interval"].PrettyPrintJsonString())
+
+        self.params.ValidateAndAssignDefaults(default_settings)
+
+        # getting the ModelPart from the Model
+        self.model_part_name = self.params["model_part_name"].GetString()
+        if self.model_part_name == "":
+            raise Exception('No "model_part_name" was specified!')
+        else:
+            self.model_part = model[self.model_part_name]
+
+    def ExecuteInitialize(self):
+
+        self.interval = KratosMultiphysics.Vector(2)
```

Comment:
If you save the parameters as member variable you don't need these to be a member variable too.

Reply:
> If you save the parameters as member variable you don't need these (`interval`, `cfl_threshold`, `format`, `output_step` and so on) to be a member variable too. If you want to have them as member variables to avoid accessing the parameters container each time, I suggest to do it in the constructor after the validation of the defaults.

Reply:
Well, as you saw from the description, this takes as an example script the compute_drag_process which also deals with these values the same way in ExecuteInitialize().

Reply:
Also the same holds for private function naming, that is why it is _EvaluateCFL() instead of _evaluate_cfl(), which would be the more canonic pythonic way.

Reply:
> Well, as you saw from the description, this takes as an example script the compute_drag_process which also deals with these values the same way in ExecuteInitialize().

We did the drag process long time ago so don't take it as reference... :laughing: 
My point is that it is redundant to save both the parameters and these variables.